File: backend/schemas/identidad.py
# schemas/identidad.py
from pydantic import BaseModel, Field
from datetime import date, datetime
from typing import Optional, List


class IdentidadBase(BaseModel):
    nombre: str = Field(..., min_length=2, max_length=100)
    apellido: str = Field(..., min_length=2, max_length=100)
    edad: Optional[int] = None
    fecha_nacimiento: Optional[date] = None
    profesion: Optional[str] = Field(None, max_length=100)
    nivel_educativo: Optional[str] = Field(None, max_length=100)
    nombre_padre: Optional[str] = Field(None, max_length=100)
    nombre_madre: Optional[str] = Field(None, max_length=100)
    numero_hermanos: Optional[int] = None
    bibliografia: Optional[str] = None
    observaciones: Optional[str] = None
    avatar: Optional[str] = None
    id_estado: Optional[int] = None
    id_genero: Optional[int] = None
    id_situacion_sentimental: Optional[int] = None
    id_orientacion_sexual: Optional[int] = None
    id_orientacion_politica: Optional[int] = None
    id_nacionalidad: Optional[int] = None
    id_pais_residencia: Optional[int] = None

class IdentidadCreate(IdentidadBase):
    pass


class IdentidadUpdate(BaseModel):
    nombre: Optional[str] = Field(None, min_length=2, max_length=100)
    apellido: Optional[str] = Field(None, min_length=2, max_length=100)
    edad: Optional[int] = None
    fecha_nacimiento: Optional[date] = None
    profesion: Optional[str] = Field(None, max_length=100)
    nivel_educativo: Optional[str] = Field(None, max_length=100)
    nombre_padre: Optional[str] = Field(None, max_length=100)
    nombre_madre: Optional[str] = Field(None, max_length=100)
    numero_hermanos: Optional[int] = None
    bibliografia: Optional[str] = None
    observaciones: Optional[str] = None
    avatar: Optional[str] = None

    id_estado: Optional[int] = None
    id_genero: Optional[int] = None
    id_situacion_sentimental: Optional[int] = None
    id_orientacion_sexual: Optional[int] = None
    id_orientacion_politica: Optional[int] = None
    id_nacionalidad: Optional[int] = None
    id_pais_residencia: Optional[int] = None


class IdentidadOut(IdentidadBase):
    id_identidad: int

    estado_nombre: Optional[str] = None
    genero_nombre: Optional[str] = None
    situacion_sentimental_nombre: Optional[str] = None
    orientacion_sexual_nombre: Optional[str] = None
    orientacion_politica_nombre: Optional[str] = None
    nacionalidad_nombre: Optional[str] = None
    pais_residencia_nombre: Optional[str] = None

    class Config:
        from_attributes = True  

File: frontend/src/api/identidad.ts
import api from './client.ts'

export interface Identidad {
    id_identidad: number;
    nombre: string;
    apellido: string;
    edad?: number | null;
    avatar?: string | null
    fecha_nacimiento?: string | null;
    profesion?: string | null;
    nivel_educativo?: string | null;
    nombre_padre?: string | null;
    nombre_madre?: string | null;
    numero_hermanos?: number | null;
    bibliografia?: string | null;
    observaciones?: string | null;
    estado_nombre?: string | null;
    genero_nombre?: string | null;
    situacion_sentimental_nombre?: string | null;
    orientacion_sexual_nombre?: string | null;
    orientacion_politica_nombre?: string | null;
    nacionalidad_nombre?: string | null;
    pais_residencia_nombre?: string | null;
    id_estado?: number | null;
    id_genero?: number | null;
    id_situacion_sentimental?: number | null;
    id_orientacion_sexual?: number | null;
    id_orientacion_politica?: number | null;
    id_nacionalidad?: number | null;
    id_pais_residencia?: number | null;

}

export const getIdentidades = async (): Promise<Identidad[]> => {
    const response = await api.get('/identidad/list');
    return response.data;
}

export const getIdentidad = async (id: number): Promise<Identidad> => {
    const response = await api.get(`/identidad/${id}`);
    return response.data;
}
export const createIdentidad = async (data: Partial<Identidad>) => {
    const res = await api.post('(identidad/crear', data);
    return res.data;
}
export const updateIdentidad = async (id: number, data: Partial<Identidad>) => {
    const res = await api.post(`/identidad/actualizar/${id}`, data);
    return res.data;
}
export const deleteIdentidad = async (id: number) => {
    await api.delete(`/identidad/eliminar/${id}`);
}